ksqldb-engine/src/main/java/io/confluent/ksql/security/KsqlPrincipal.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.security;
+
+import java.security.Principal;
+import java.util.Map;
+
+public interface KsqlPrincipal extends Principal {
```

Comment:
This class should be used instead of `Principal` pretty much everywhere in the ksql codebase, right? If so, we should probably document that in the javadoc.

Reply:
Sure, I'll add a javadoc.